include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -369,7 +369,12 @@ class gap_decorator_anchor_set
     ~gap_decorator_anchor_set() = default;
 
     //!\brief Construct with the ungapped range type.
-    constexpr gap_decorator_anchor_set(inner_type const & range) : ungapped_view{view::all(range)} {}
+    template <typename other_range_t>
+         requires !std::Same<other_range_t, gap_decorator_anchor_set> &&
+                  std::Same<remove_cvref_t<other_range_t>, remove_cvref_t<inner_type>> &&
+                  std::ranges::ViewableRange<other_range_t> // at end, otherwise it competes with the move ctor
+    gap_decorator_anchor_set(other_range_t && range) : ungapped_view{view::all(std::forward<inner_type>(range))}
+    {} // TODO (@smehringer) why does this not work with other_range_t ---------------------------^
```

Comment:
because you forgot to add const?

Reply:
Since you are now holding `const &` to all non-views (something I am pretty sure we need to change in the long run 😢 ), can't you just do:
```cpp
gap_decorator_anchor_set(inner_type const & range) : ungapped_view{view::all(range)} {}
gap_decorator_anchor_set(inner_type && range) = delete;
```
?

Reply:
Ah, no, this covers initialisation from view and non-view? This whole thing is even more complex than the views 🤦‍♂️ 

I really think we should change it where the inner_type is the template parameter. Then we can model everything similar to views...

Anyway, what you need up there is to add conditional const-ness inside the cast. But only for non-views of course. It will be ugly. It might be easier to make separate constructors for view and non-view. Or just change the design as mentioned above straight away.

Reply:
But everything works like this :D can't we leave it for now?

Reply:
I think this overload is always preferred over the other one, also for view-temporaries. Since you cast to inner_type, you will invoke a copy operation of the view at that point. This will break for views that are move-only.
If you cannot come up with a fix now, please clearly document this behaviour and add a card for changing it later on.

Maybe we can sit together some time this week and make a plan for the ranges/view/decorator stuff that needs to be done and in which order.

Reply:
Is there a move only view I can test?

I would document it for now and add a cart because this PR blocks alignment file input and we need this next week. Is that fine?

Reply:
And yes we should make a plan for the views.

Reply:
> Is there a move only view I can test?

No, ATM copyability is still required, but this has been changed so some views, e.g. istream_range or our files will be move-only.

> I would document it for now and add a card because this PR blocks alignment file input and we need this next week. Is that fine?

Yep!

Reply:
Done